client/components/profilePage/allUsersModal.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Image, Button, Table, Badge, Spinner, Row, Col } from 'react-bootstrap';
+import axios from 'axios';
+
+export default function AllUsers({
+  handleClose,
+  show,
+}) {
+
+  const [allUsers, setAllUsers] = useState([]);
+
+  useEffect(async () => {
+    let usersList = await axios.get(`http://localhost:3000/user/all`)
+    setAllUsers(usersList.data);
+  }, []);
+
+  return allUsers === undefined ? (
+    <Spinner animation='border' role='status'>
```

Comment:
same as the above comment. feels unnecessary to have the undefined part, unless there is some issue that necessitates it